backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/policyunits/NumaPolicyUnit.java

```diff
@@ -48,7 +49,7 @@ public List<VDS> filter(SchedulingContext context, List<VDS> hosts, List<VM> vmG
         }
 
         List<VM> vmsToCheck = vmGroup.stream()
-                .filter(vm -> !vm.getvNumaNodeList().isEmpty())
+                .filter(vm -> !vm.getvNumaNodeList().isEmpty() || VmCpuCountHelper.isResizeAndPinPolicy(vm))
```

Comment:
ok, so iiuc this change and the one above were done just to get the check of host.isNumaSupport() later on, right? while the change above makes some sense (to consider a vm with resize-and-pin policy as vm with numa pinning), this one is hard for me to parse.. so if we have, let's say, a single vm without numa nodes and with resize-and-pin policy then vmsToCheck will include it and we'll get to the check of host.isNumaSupport() but we'll also call NumaPinningHelper.findAssignment which will return 'empty', no?

Reply:
> ok, so iiuc this change and the one above were done just to get the check of host.isNumaSupport() later on, right? 

right.
> while the change above makes some sense (to consider a vm with resize-and-pin policy as vm with numa pinning), this one is hard for me to parse.. so if we have, let's say, a single vm without numa nodes and with resize-and-pin policy then vmsToCheck will include it and we'll get to the check of host.isNumaSupport() but we'll also later call NumaPinningHelper.findAssignment which will return 'empty', no?

The first `vmNumaPinned` is the fastest check to skip the policy unit. Once we pass it, it checks again the VMs and we need to include the `resize` VMs there. If it's before we set the vNUMA configuration (e.g initial run of such VM), 
`NumaPinningHelper.findAssignment()` will return `Optional.empty()`. `isPresent()` checks if it's null, not empty.

Reply:
> The first `vmNumaPinned` is the fastest check to skip the policy unit. Once we pass it, it checks again the VMs and we need to include the `resize` VMs there. If it's before we set the vNUMA configuration (e.g initial run of such VM), `NumaPinningHelper.findAssignment()` will return `Optional.empty()`. `isPresent()` checks if it's null, not empty.

the problem is that while `Optional.empty().isPresent()` returns false, we'll filter out the host below

Reply:
Yes, I probably missed that when testing, I think I checked the policy "until" numa support.
Now I clear the list from VMs with empty (like we do in the start of the policy).